fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -464,6 +467,11 @@ public void dropDb(String dbName, boolean isForceDrop) throws DdlException, Meta
                 throw new MetaNotFoundException("Database not found");
             }
             db = this.fullNameToDb.get(dbName);
+            if (!isForceDrop && db.getTemporaryTables().size() > 0) {
+                throw new DdlException("The database [" + dbName + "] " +
```

Comment:
why cannot drop db if tmp tables remain in?

Reply:
> why cannot drop db if tmp tables remain in?

For safety reasons, dropping the database will cause subsequent access failures to the temporary table, so a prompt message will be given here. If the user is aware of the consequences, he or she can use drop force to delete it.

Reply:
> but for normal table, we can drop the db(not force) without prompt?

in some cases we can't drop the db too.
![image](https://github.com/StarRocks/starrocks/assets/3675229/d78745ed-2ffa-4427-baff-d70c6a4c6400)